zerver/views/invite.py

```diff
@@ -44,25 +45,22 @@ def check_role_based_permissions(
 
 
 @require_member_or_admin
-@has_request_variables
+@typed_endpoint
 def invite_users_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    invitee_emails_raw: str = REQ("invitee_emails"),
-    invite_expires_in_minutes: int | None = REQ(
-        json_validator=check_none_or(check_int), default=INVITATION_LINK_VALIDITY_MINUTES
-    ),
-    invite_as: int = REQ(
-        json_validator=check_int_in(
-            list(PreregistrationUser.INVITE_AS.values()),
-        ),
-        default=PreregistrationUser.INVITE_AS["MEMBER"],
-    ),
-    notify_referrer_on_join: bool = REQ(
-        "notify_referrer_on_join", json_validator=check_bool, default=True
-    ),
-    stream_ids: list[int] = REQ(json_validator=check_list(check_int)),
-    include_realm_default_subscriptions: bool = REQ(json_validator=check_bool, default=False),
+    *,
+    invitee_emails_raw: Annotated[str, ApiParamConfig("invitee_emails")],
+    invite_expires_in_minutes: Json[int | None] = INVITATION_LINK_VALIDITY_MINUTES,
+    invite_as: Annotated[
+        Json[int],
+        check_int_in_validator(list(PreregistrationUser.INVITE_AS.values())),
+    ] = PreregistrationUser.INVITE_AS["MEMBER"],
+    notify_referrer_on_join: Annotated[
+        Json[bool], ApiParamConfig("notify_referrer_on_join")
+    ] = True,
```

Comment:
Is `ApiParamConfig` a noop here?

Reply:
Ah , yes. The parameter name was specified earlier, so I assumed the name was different. I'll change it.